databuilder/models/badge.py

```diff
@@ -18,6 +18,10 @@ def __repr__(self) -> str:
         return 'Badge({!r}, {!r})'.format(self.name,
                                           self.category)
 
+    def __eq__(self, other: Any) -> bool:  # type: ignore[override]
```

Comment:
other: Badge

Reply:
Oh yeah of course! Thanks

Reply:
Hmm, apart from that it then throws an error

```python
    def __eq__(self, other: Badge) -> bool:
E   NameError: name 'Badge' is not defined
```

Apparently the Class itself isn't know at that point - which I find a little strange. So I'm keeping `Any` unless you know a trick.

I got rid if the `  # type: ignore[override]` though 🙂 

--- 

Oh wait, through [this](https://coderoad.ru/54801832/mypy-__eq__-%D0%BD%D0%B5%D1%81%D0%BE%D0%B2%D0%BC%D0%B5%D1%81%D1%82%D0%B8%D0%BC%D0%BE-%D1%81-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%BC-supertype) i found out it needs to be  `other: 'Badge'` and then mypy itself gives the full recommended way:

```python
❯ mypy .
databuilder/models/badge.py:21: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
databuilder/models/badge.py:21: note: This violates the Liskov substitution principle
databuilder/models/badge.py:21: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
databuilder/models/badge.py:21: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
databuilder/models/badge.py:21: note:     def __eq__(self, other: object) -> bool:
databuilder/models/badge.py:21: note:         if not isinstance(other, Badge):
databuilder/models/badge.py:21: note:             return NotImplemented
databuilder/models/badge.py:21: note:         return <logic to compare two Badge instances>
```

Reply:
wow that's so useful